### PR TITLE
[Web Portal] Display error message in job detail page

### PR DESCRIPTION
**Pull request by sunqinzheng**

### CODE REVIEW

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx [hunk 1]
```diff
@@ -67,13 +73,73 @@ export default class Summary extends React.Component {
...
+        // user code error
+        let userExitCode;
+        if (diag) {
+          let match = diag.match(/<Raw>\[ExitCode\]: (\d+)/);
+          if (match) {
+            userExitCode = parseInt(match[1], 10);
+          }
+        }
+        if (!isNil(userExitCode)) {
+          return `Failed due to user error. Exit Code: ${userExitCode}. Please check stderr and stdout for more information.`;
```
Comment (scarlett2018) [inline src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx]: **Error Type**: User Error;
**Error code**: 1;
**Error log(s)**: please check Stdout and Stderr for each task.

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx [hunk 2]
```diff
@@ -67,13 +73,73 @@ export default class Summary extends React.Component {
...
+        } else {
+          return `Failed due to user error. Please check container's stderr and stdout for more information.`;
```
Comment (scarlett2018) [inline src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx]: **Error Type**: User Error;
**Error log(s)**: please check Stdout and Stderr for each task.

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx [hunk 3]
```diff
@@ -67,13 +73,73 @@ export default class Summary extends React.Component {
...
+        }
+      } else {
+        return (
+          <div>
+            <span>Failed due to platform error. Please send the </span>
+            <Link onClick={this.showApplicationSummary}>application summary</Link>
```
Comment (scarlett2018) [inline src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx]: **Error Type**: System Error;
**Error log(s)**: please send the [system log](this.showApplicationSummary) to your administrator for further investigation.

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx [hunk 4]
```diff
@@ -67,13 +73,73 @@ export default class Summary extends React.Component {
...
+            <span> to your admin</span>
+          </div>
+        );
+      }
+    } else if (state === 'Waiting') {
+      const resourceRetries = get(jobInfo, 'jobStatus.retryDetails.resource');
+      if (resourceRetries >= 3) {
+        return (
+          <div>
+            <span>Resource allocation failed more than 3 times. Please check the resource requirement in your</span>
```
Comment (scarlett2018) [inline src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx]: let's discuss with @yqwang-ms about this.

**Comment by sunqinzheng** [comment]
图片

**Pull request merged**